amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/FlinkKinesisConsumer.java

```diff
@@ -462,6 +465,25 @@ public void snapshotState(FunctionSnapshotContext context) throws Exception {
 		}
 	}
 
+	/**
+	 * Creates state serializer for kinesis shard sequence number.
+	 * Using of the explicit state serializer with KryoSerializer is needed because otherwise
+	 * users cannot use 'disableGenericTypes' properties with KinesisConsumer, see FLINK-24943 for details
+	 *
+	 * @return state serializer
+	 */
+	@VisibleForTesting
```

Comment:
`@VisibleForTesting` is a smell we should try to avoid. If this method needs to be visible for testing I would recommend moving it to a dedicated class with well defined responsibility and making public.